firebase-database/src/main/java/com/google/firebase/database/core/Repo.java

```diff
@@ -529,19 +530,18 @@ public void onComplete(@NonNull Task<Object> task) {
                                   query.getRef(),
                                   IndexedNode.from(serverNode, query.getSpec().getIndex())));
                         }
+                        scheduleNow(
+                            new Runnable() {
+                              @Override
+                              public void run() {
+                                serverSyncTree.setQueryInactive(query.getSpec());
+                              }
+                            });
                       }
                     });
           }
         });
-    return source
```

Comment:
I think you could keep this as is and just pass the executor that `scheduleNow` uses as the first argument to `addOnCompleteListener`. That would be a bit easier to follow in my opinion.

Reply:
I don't think the executor service in the DefaultRunLoop is available here:

https://github.com/firebase/firebase-android-sdk/blob/master/firebase-database/src/main/java/com/google/firebase/database/core/utilities/DefaultRunLoop.java#L99

Since it's hidden behind the RunLoop interface:

https://github.com/firebase/firebase-android-sdk/blob/master/firebase-database/src/main/java/com/google/firebase/database/core/RunLoop.java#L24

This feels like more trouble than it's worth for marginal readability gains. Let me know if I missed something.